proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,636 @@
+
+\begin{codeblock}
+template <InputRange Rng, NoThrowForwardIterator O>
+  requires Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
+    uninitialized_copy(Rng&& rng, O result);
+\end{codeblock}
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        return uninitialized_copy(begin(rng), end(rng), result);
+\end{codeblock}
+
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[begin(rng), end(rng))}.
+
+\begin{codeblock}
+template <InputIterator I, NoThrowForwardIterator O>
+  requires Constructible<value_type_t<O>, reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)>
+    uninitialized_copy_n(I first, difference_type_t<I> n, O result);
+\end{codeblock}
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        return uninitialized_copy(make_counted_iterator(first, n),
+                                  default_sentinel{}).base();
+\end{codeblock}
+
+\requires \newtxt{\tcode{result} shall not be in the range \tcode{make_counted_iterator(first, n)}.}
+} %% \color{addclr}
+
+\rSec3[uninitialized.move]{\tcode{uninitialized_move}}
+{\color{remclr}
+\begin{codeblock}
+template <class InputIterator, class ForwardIterator>
+  ForwardIterator uninitialized_move(InputIterator first, InputIterator last,
+                                     ForwardIterator result);
+\end{codeblock}
+
+\begin{codeblock}
+template <class InputIterator, class Size, class ForwardIterator>
+  pair<InputIterator, ForwardIterator>
+    uninitialized_move_n(InputIterator first, Size n, ForwardIterator result);
+\end{codeblock}
+
+\setcounter{Paras}{2}
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        for (; n > 0; ++result, (void) ++first, --n)
+          ::new (static_cast<void*>(addressof(*result)))
+            typename iterator_traits<ForwardIterator>::value_type(std::move(*first));
+        return {first,result};
+\end{codeblock}
+} %% \color{remclr}
+
+{\color{addclr}
+\begin{codeblock}
+template <InputIterator I, Sentinel<I> S, NoThrowForwardIterator O>
+  requires Constructible<value_type_t<O>, rvalue_reference_t<I>>() &&
+           __ReferenceTo<O, value_type_t<O>>()
+    tagged_pair<tag::in(I), tag::out(O)> uninitialized_move(I first, S last, O result);
+\end{codeblock}
+} %% \color{addclr}
+
+\setcounter{Paras}{0}
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        for (; first != last; (void)++result, ++first)
+          ::new @\added{(const_cast<void*>}@(static_cast<@\added{const volatile }@void*>(addressof(*result)))@\added{)}@
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(std::move(*first));
+        return @\changed{result}{\{first, result\}}@;
+\end{codeblock}
+
+\pnum
+\remarks If an exception is thrown, some objects in the range \tcode{[first, last)} are left in a valid but
```

Comment:
C++ draft latex shortcut: `\tcode{[foo, bar)}` is the macro expansion of `\range{foo}{bar}`. It's not worth the trouble of changing what you have here, but for future reference.